embrace-android-delivery/src/test/kotlin/io/embrace/android/embracesdk/internal/delivery/intake/IntakeServiceImplTest.kt

```diff
@@ -188,14 +190,14 @@ class IntakeServiceImplTest {
             latch.countDown()
         }
         worker.shutdownAndWait(1000)
-        assertEquals(8, payloadStorageService.storedFilenames.size)
+        assertEquals(4, payloadStorageService.storedPayloadCount())
```

Comment:
This indicates a potential bug to me - the intake service has stored 8 payloads with different filenames but the payload storage service only seems to have 4?

Reply:
The file names are based on the metadata, aren't they? So this results in 4 writes and 4 overwrites.

Or is the intention to keep track of the all the versions of the saved payloads?

I think in my version, I was trying to emulate the real file system but in memory. 

I think we can have both if we want. Let me ping you and we can figure out what to do.